guides/ci/travis.md

```diff
@@ -0,0 +1,252 @@
+
+### Using a specific Crystal release
+
+So, let's suppose we want to use version [0.31.1](https://github.com/crystal-lang/crystal/releases/tag/0.31.1) of the Crystal compiler. For this we are going to use [Docker](https://www.docker.com/). First we need to add Docker as a service in `.travis.yml`, and then we can use `docker` commands in our build steps, like this:
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+script:
+  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec
+```
+
+**Note:** A list with the different official [Crystal docker images](https://hub.docker.com/r/crystallang/crystal/tags) is available at [DockerHub](https://hub.docker.com/r/crystallang/crystal).
+
+### Using `latest`, `nightly` and a specific Crystal release all together!
+
+For using 3 versions of the compiler we are going to use a different configuration file based on [Build Matrix](https://docs.travis-ci.com/user/customizing-the-build#build-matrix)
+Here is the example:
+
+```yaml
+# .travis.yml
+matrix:
+ include:
+   - language: crystal
+     crystal:
+       - latest
+     script:
+       - crystal spec
+
+   - language: crystal
+     crystal:
+       - nightly
+     script:
+       - crystal spec
+
+   - language: bash
+     services:
+       - docker
+     script:
+       - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec
+```
+
+## Installing shards packages
+
+Thanks to the `language: crystal` support, Travis CI automatically runs `shards install`. And for faster ci times we may use [caching](#caching)
+
+#### Using Docker
+
+When using a containerization service like Docker we will need to run `shards install` explicitly, like this:
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+script:
+  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 shards install
+  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec
+```
+
+**Note:** Since the shards will be installed in `./lib/` folder, it will be preserved for the second docker run command.
+
+## Installing binary dependencies
+
+> It is important to mention that there are many ways to achieve this and **it will heavily depend on our development workflow**.
+
+Our application or maybe some shards may required libraries and packages. In that case we will need to install them using `apt` (or similar).
+
+Here is a first example installing SQLite3 using Travis CI configuration file:
+
+```yaml
+# .travis.yml
+language: crystal
+crystal:
+  - latest
+
+before_install:
+  - sudo apt-get -y install libsqlite3-dev
+
+addons:
+  apt:
+    update: true
+
+script:
+  - crystal spec
+```
+
+#### Using Docker
+
+The same can be accomplished using a containerization service like Docker. Here is an example of a `Dockerfile` installing SQLite3:
```

Comment:
Using a `Dockerfile` is a very different method and introduces a completely new concept without much context. An example more similar to the previous section using `docker run` to install a dependency in the container is probably not a good idea, because it is more complex. But maybe that should be mentioned here? And a few more words about the method `Dockerfile`? When using a `Dockerfile` anyway, it would also make sense to install shards there.

However, a `Dockerfile` for setting up dependencies is not specific to travis CI, so maybe there should rather be a separate guide for more details which would be referenced here.